flocker/common/_interface.py

```diff
@@ -54,3 +54,27 @@ def class_decorator(cls):
             setattr(cls, name, method_decorator(name, *args, **kwargs))
         return cls
     return class_decorator
+
+
+def provides(interface):
+    """
+    Create an invariant that asserts that the given value provides the given
+    interface.
+
+    :param InterfaceClass interface: The interface to check for.
+
+    :return: A function that takes an object and returns a tuple of `bool` and
+        a error message.
+    """
+    interface_name = interface.__name__
+
+    def invariant(value):
+        if interface.providedBy(value):
```

Comment:
:grey_question: Is there a reason not to just do:

```
return (interface.providedBy(value), "{value!r} doesn't provide {interface}".format(
            value=value, interface=interface_name,
        ))
```

The API seems to suggest that's the more standard way to do it:
http://pyrsistent.readthedocs.org/en/latest/intro.html#invariants